### DISCUSSION TITLE
Svg Image object binding.

### DISCUSSION THREAD

**Discussion by Alexbeerses** [wieslawsoltes/Svg.Skia#254] [Q&A]
Good day!

I need to bind svg image object in avalonia xaml with Community MVVM. 
I need to bind image object not image path because image comes from far source in my project.
The button pressing should make image changing in runtime. I tried to use SvgSource but it didn't work.
I have make a simple project to show the behavior. 

[AvaloniaSvg.zip](https://github.com/user-attachments/files/18208397/AvaloniaSvg.zip)

Could you help?

```xaml
    <StackPanel
        HorizontalAlignment="Center"
        VerticalAlignment="Center"
        Orientation="Vertical">
        
        <Image Width="128">
            <Image.Source>
                <SvgImage Source="{Binding StateImage}" />
            </Image.Source>
        </Image>

        <Button HorizontalAlignment="Center" Command="{Binding StateImageChange}">Change</Button>

    </StackPanel>
```

```cs
public partial class MainWindowViewModel : ViewModelBase
{
    //-----------------------------------------------------------------------------------------------

    const string SvgPathPrefix = "avares://AvaloniaSvg/Assets/";

    //-----------------------------------------------------------------------------------------------

    [ObservableProperty]
    SvgSource? _StateImage = SvgImageGet("BallRed");

    //-----------------------------------------------------------------------------------------------

    [RelayCommand]
    public void StateImageChange()
    {
        StateImage = SvgImageGet("BallGreen");
    }
    //-----------------------------------------------------------------------------------------------

    public static SvgSource? SvgImageGet(string name)
    {
        var path = $"{SvgPathPrefix}{name}.svg";

        try
        {
            return SvgSource.Load(path);
        }
        catch (Exception ex)
        {
            string s = ex.Message;
        }
        return null;
    }
    //-----------------------------------------------------------------------------------------------
}
```

**Comment by KarenArzumanyan** [comment]
This problem creates a lot of inconvenience and difficulty.
In cases where an image is loaded from a separate library, using a path does not provide a solution.
Please help us fix it

**Reply by wieslawsoltes** [reply]
Please use Svg control instead of Image it can bind to paths. SvgImage doesn't have invalidation in Avalonia

**Reply by Alexbeerses** [reply]
> Please use Svg control instead of Image it can bind to paths. SvgImage doesn't have invalidation in Avalonia

It is neccessary to use Image (not Svg control) for example for avalonia MenuItem:

```cs
[ObservableProperty]
public SvgSource? _Image;
```

```xml
<UserControl.Styles>
<Style Selector="MenuItem">
<Setter Property="Icon">
    <Template>
        <Image Width="32" Height="32">
            <Image.Source>
                <SvgImage Source="{Binding Image}" />
            </Image.Source>
        </Image>
    </Template>
</Setter>
</Style>
</UserControl.Styles>
```

**Reply by wieslawsoltes** [reply]
Please sent request here otherwise I can't do anything as it's external dependency issue https://github.com/AvaloniaUI/Avalonia/issues/17601

**Reply by Alexbeerses** [reply]
> Please sent request here otherwise I can't do anything as it's external dependency issue [AvaloniaUI/Avalonia#17601](https://github.com/AvaloniaUI/Avalonia/issues/17601)

Ok thank you. I'll do.

**Reply by Alexbeerses** [reply]
Done.
